File: packages/ui/src/flat-picker.test.ts
import { describe, expect, test } from "bun:test";
import {
  firstFlatIdInWing,
  flatsInWing,
  uniqueWingNames,
  wingForFlatId,
  wingKey,
  wingLabel,
} from "./flat-picker";

const flats = [
  { id: "2", number: "201", wingName: "B" },
  { id: "1", number: "101", wingName: "A" },
  { id: "3", number: "102", wingName: "A" },
  { id: "4", number: "G1", wingName: "  " },
];

describe("flat-picker", () => {
  test("lists unique wings and flats in the chosen wing", () => {
    expect(uniqueWingNames(flats)).toEqual(["", "A", "B"]);
    expect(flatsInWing(flats, "A").map((f) => f.number)).toEqual(["101", "102"]);
    expect(firstFlatIdInWing(flats, "B")).toBe("2");
    expect(wingForFlatId(flats, "3")).toBe("A");
    expect(wingKey({ wingName: " A-C " })).toBe("A-C");
    expect(wingLabel("")).toBe("No wing");
    expect(wingLabel("A")).toBe("A");
  });
});

File: apps/mobile/test/unit/flat_picker_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:societyhub_mobile/api/models.dart';
import 'package:societyhub_mobile/shared/flat_picker.dart';

FlatDto _f({
  required String id,
  required String number,
  String? wingName,
}) {
  return FlatDto(id: id, number: number, wingName: wingName);
}

void main() {
  test('lists unique wings and flats in the chosen wing', () {
    final flats = [
      _f(id: '2', number: '201', wingName: 'B'),
      _f(id: '1', number: '101', wingName: 'A'),
      _f(id: '3', number: '102', wingName: 'A'),
    ];
    expect(uniqueWingNames(flats), ['A', 'B']);
    expect(flatsInWing(flats, 'A').map((f) => f.number), ['101', '102']);
    expect(firstFlatIdInWing(flats, 'B'), '2');
    expect(wingForFlatId(flats, '3'), 'A');
    expect(wingLabel(''), 'No wing');
  });
}
